contrib/inventory/hyperone.py

```diff
@@ -0,0 +1,215 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+from __future__ import print_function
+from time import time
+
+import requests
+import os
+import sys
+import argparse
+import json
+
+try:
+    import ConfigParser as configparser
+except ImportError:
+    import configparser
+'''
+External inventory script for HyperOne
+====================================
+
+Contains parts shamelessly copied from an existing inventory script.
+
+Generates an inventory that Ansible can understand by making API requests
+to HyperOne API. Each VM tag match Ansible host group.
+
+Before using this script you need create  hyperone.ini config file and put
+token to access service.
+
+To generate token you can use h1-cli and following commands:
+
+$ TOKEN_ID=$(h1 project token add --name ansible-inventory --query "[0]._id" -o json);
+$ h1 project token access delete --token $TOKEN_ID --yes \
+    --access $(h1 project token access list --token $TOKEN_ID -o tsv --query "[].[_id]" );
+$ h1 project token access add --token $TOKEN_ID --method GET --path '/vm'
+$ echo "Token ID: $TOKEN_ID";
+
+Example config file:
+
+    [hyperone]
+    token=$TOKEN_ID
+    cache_path=~/.cache/hyperone
+'''
+
+
+class HyperOneAPI(object):
+    def __init__(self, token):
+        self.session = requests.Session()
+        self.session.headers = {'X-Auth-Token': token}
+
+    def get_vms(self):
+        return self.session.get('https://api.hyperone.com/v1/vm').json()
+
+    def get_vm(self, vm_id):
+        return self.session.get("https://api.hyperone.com/v1/vm/{vm_id}".format(vm_id=vm_id)).json()
+
+    def get_vm_netadp(self, vm_id):
+        return self.session.get("https://api.hyperone.com/v1/vm/{vm_id}/netadp".format(vm_id=vm_id)).json()
+
+
+class FileCache(object):
+    def __init__(self, filename, max_age):
+        self.file = filename
+        self.max_age = max_age
+
+    def is_valid(self):
```

Comment:
Also, it's a classical property. Refactor it in a more elegant way:
```python
@property
def valid(self):
     try:
         return os.path.getmtime(self.file) + self.max_age > time()
     except IOError:
         return False
```

P.S. When reusing this snippet, please add the trailer to the long commit description:

```
Co-authored-by: Sviatoslav Sydorenko <wk@sydorenko.org.ua>
```

Reply:
Fixed!